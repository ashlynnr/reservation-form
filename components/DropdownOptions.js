import React from "react";
import styled from "styled-components";

const DropdownOptions = ({
  title,
  ages,
  currentValue,
  handleChange,
  guestType,
  checked,
  options
}) => {
  return (
    <Section>
      {title} <br />
      {ages}
      <Dropdown
        value={currentValue ? currentValue : 1}
        onChange={e => handleChange(e, guestType)}
      >
        {options.map((guests, index) => (
          <option key={guests + index} value={guests} disabled={!checked}>
            {guests}
          </option>
        ))}
      </Dropdown>
    </Section>
  );
};

export default DropdownOptions;

const Section = styled.div`
  font-size: 1.1em;
  display: grid;
  grid-gap: 5px;
  align-items: center;
`;
const Dropdown = styled.select`
  width: 50px;
  font-size: 1.05em;
`;
