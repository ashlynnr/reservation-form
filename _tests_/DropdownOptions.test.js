import React from "react";
import { shallow } from "enzyme";
import rooms from "../utils/data";
import DropdownOptions from "../components/DropdownOptions";
import { jestPreset } from "ts-jest";

describe("Dropdown Options", () => {
  const wrapper = shallow(
    <DropdownOptions
      title={"unicorns"}
      ages={"20-30"}
      currentValue={1}
      handleChange={jest.fn}
      guestType={"adults"}
      checked
      options={[1, 2, 3]}
    />
  );

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders only one dropdown", () => {
    const dropdown = wrapper.find("DropdownOptions__Dropdown");

    expect(dropdown.length).toEqual(1);
  });

  it("changes the value when selected", () => {
    const dropdown = wrapper.find("DropdownOptions__Dropdown");
    dropdown.simulate("change", { target: { value: 1 }, key: "adults" });

    expect(dropdown.props().value).toBe(1);
  });
});
