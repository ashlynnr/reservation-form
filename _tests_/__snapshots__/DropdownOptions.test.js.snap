// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Options matches snapshot 1`] = `
<DropdownOptions__Section>
  unicorns
   
  <br />
  20-30
  <DropdownOptions__Dropdown
    onChange={[Function]}
    value={1}
  >
    <option
      disabled={false}
      key="1"
      value={1}
    >
      1
    </option>
    <option
      disabled={false}
      key="3"
      value={2}
    >
      2
    </option>
    <option
      disabled={false}
      key="5"
      value={3}
    >
      3
    </option>
  </DropdownOptions__Dropdown>
</DropdownOptions__Section>
`;
